src/services/utils/get-sorted-dimension-info.ts

```diff
@@ -0,0 +1,26 @@
+import { PSEUDO_DIMENSION_INDEX } from "../../constants";
+import type { ExtendedDimensionInfo, PivotLayout, PseudoDimension } from "../../types/QIX";
+
+const filterPseudoAndVisibleDimensions = (qDim: ExtendedDimensionInfo | -1) =>
+  qDim === PSEUDO_DIMENSION_INDEX || qDim.qCardinalities.qHypercubeCardinal > 0;
```

Comment:
just a question out of curiosity:  
Is `qDim.qCardinalities.qHypercubeCardinal > 0` the way we detect a normal dimension?
On the flip side, could it be any other dimension type rather than normal and pseudo?

Reply:
In  this logic `qDim.qCardinalities.qHypercubeCardinal > 0` is used as a way to check if a dimension is visible to the user. By visible I mean that a node in it's "parent dimension" must have at least 1 node expanded.

Ex: you have 2 dimensions on the left side. The first dimension does not have any nodes expanded. This means that the second dimension is not visible to the user. But that also means that `qDim.qCardinalities.qHypercubeCardinal > 0` is `false` for the second dimension.